mm2src/mm2_main/src/lp_swap/taker_swap_v2.rs

```diff
@@ -432,6 +434,9 @@ impl<MakerCoin: MmCoin + MakerCoinSwapOpsV2, TakerCoin: MmCoin + TakerCoinSwapOp
 
     fn taker_payment_locktime(&self) -> u64 { self.started_at + self.lock_duration }
 
+    #[inline]
+    fn spend_maker_payment_conf_timeout(&self) -> u64 { self.started_at + 4 * self.lock_duration }
```

Comment:
fixed name https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/2d80017a507ca3ef8a6ce88a2f4a5a15a62565a6

I took `3 * self.lock_duration` for maker to spend taker payment, as this is the maximum time on taker (funding) locktime.
And `4 * ` on taker side to spend maker payment, as taker spends some time searching for maker's transaction to extract secret.
These functions are supposed to be used on `PaymentSpent` state to be sure that trader got their payment and spend tx is confirmed. So I think final conf timeouts should have big multipliers.  Do you think we should increase constants?

Reply:
let me first mention the lock times of each swaps tx since these confuse me:
- `funding`       -> `3T`
- `maker payment` -> `2T`
- `taker payment` -> `1T`
- `taker payment spend` - no locktime
- `maker payment spend` - no locktime

So.

If the `taker payment spend` doesn't wanna confirm in time, as a maker, you want to refund your own `maker payment` ASAP, so after `2T` (given that it wasn't even already spent, which is very unlikely).
Same holds for the `maker payment spend`. you, as a taker, would want to refund your `taker payment` ASAP if the `maker payment spend` is stuck (given that it wasn't even already spent, which is much more unlikely since the taker knew the secret from the spend of the `taker payment` in the first place).

let me spawn @shamardy here in this case since these locktimes scare me :)

Reply:
after writing that, im not even sure if waiting for these txs to be confirmed makes sense anymore haha.
these locktimes are part of the atomicity guarantees of the atomic swap. they are in place to make sure you are able to spend and get your spend confirmed before the other party can initiate a refund.
if we can't confirm within these locktimes, we are ... just screwed :)

Reply:
> if we can't confirm within these locktimes, we are ... just screwed :)

Could you clarify what do mean by "locktimes". For confirmations we have `_conf_timeout` functions.

As for PeymentSpent state, if confirmation time is out there is still a chance that trader could spent payment on `TakerPaymentRefundRequired/MakerPaymentRefundRequired`. We are going to have smth similar you do on legacy. Try to spend or refund. Should be fine.

Well I think `fn maker_payment_conf_timeout(&self) -> u64 { self.started_at + self.lock_duration / 3 }` could be bigger

Reply:
btw this is how we calc locktime https://github.com/KomodoPlatform/komodo-defi-framework/blob/2d80017a507ca3ef8a6ce88a2f4a5a15a62565a6/mm2src/mm2_main/src/lp_swap.rs#L782-L793

Reply:
This relates to what I and @Alrighttt mentioned before, taker should act selfishly, same for maker. They should always try refund or spend/confirm endlessly until one of them works, refunds also need confirmations. Maybe refunds can start to be tried after lock time (1T) has passed but with refunds they need to check confirmations/spend as well like how we are currently fixing for legacy swaps here https://github.com/KomodoPlatform/komodo-defi-framework/pull/2280

Reply:
I would rename state to `TakerPaymentRefundOrSpendRequired` then.

> We can wait for confirmations for 1T and `TakerPaymentRefundRequired` means that we move to a state of trying to spend or refund like how it's being fixed in legacy swaps.

> We might need to implement/modify this to a different type of state machine that is called [nondeterministic finite automata](https://stackoverflow.com/a/37092774) (NFAs) where the current state is a set, so refunding and confirmation of refunding is tried while spending and confirmation of spending is tried at the same time and we move to the next state if either of those is successful.

Yes, we can implement one of these approaches (probably its better to just follow the first one in the current state machine) as the next step after #2280 merge